src/API/_compat.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\API;
+
+use function class_exists;
+use Composer\InstalledVersions;
+use Composer\Semver\VersionParser;
+
+if (
+    // Provide a backwards-compatible Psr3 implementation when running under composer.
+    class_exists(InstalledVersions::class, false)
+    && InstalledVersions::satisfies(new VersionParser(), 'composer/composer', '~2.0')
```

Comment:
Should check whether we are running as composer command first; `VersionParser` might not be available otherwise.